src/main/java/hudson/plugins/emailext/plugins/content/FailedTestsContent.java

```diff
@@ -143,6 +134,20 @@ private SummarizedTestResult prepareSummarizedTestResult(AbstractTestResultActio
         return result;
     }
 
+    private int addTest(SummarizedTestResult result, int printSize, TestResult failedTest) {
+        String stackTrace = showStack ? (escapeHtml ? escapeHtml(failedTest.getErrorStackTrace()) : failedTest.getErrorStackTrace()) : null;
+        String errorDetails = showMessage ? (escapeHtml ? escapeHtml(failedTest.getErrorDetails()) : failedTest.getErrorDetails()) : null;
+        String name = String.format("%s.%s", (failedTest instanceof CaseResult) ? ((CaseResult) failedTest).getClassName() : failedTest.getFullName(),
+                failedTest.getDisplayName());
+        FailedTest t = new FailedTest(name, failedTest.isPassed(), errorDetails, stackTrace);
+        String testYaml = t.toString();

```

Comment:
I still feel that this adds a lot of overhead (extra encoding for both YAML and text, the later being a regression in performance). Like I said in my previous comment, I prefer coming up with a heuristic number instead of doing so much more computation to come up with an accurate number. What do you think @slide ?

Reply:
I don't think the regression in performance is going to be very noticeable. I think an accurate number is important.